lib/system/database.rb

```diff
@@ -125,6 +124,40 @@ def by_id_or_system_name(id_or_system_name)
       end
     end
 
+    module Definitions
+      extend ActiveSupport::Concern
+
+      included do
+        @triggers = []
+        @procedures = []
+
+        class << self
+          attr_reader :triggers, :procedures
+
+          def define(&block)
+            @triggers.clear
+            @procedures.clear
+
+            class_eval &block
+          end
+
+          def trigger(name, options = {})
+            klass_name = "System::Database::#{ancestors.first.to_s.demodulize}::Trigger"
```

Comment:
We should only load the needed database definition depending on the `ENV['DB']` and load only the necessary classes. Right now with `rails r 'puts System::Database::Postgres'` works with MySQL.